### DISCUSSION TITLE
Tree shaking problem

### DISCUSSION THREAD

**Discussion by fzyzcjy** [ethanblake4/flutter_eval#16] [General]
In order to support *all* flutter api, we seem to need to finally bridge all of them. Then, Flutter tree shaking is effectively disabled, causing the size to be larger.

This may not be a problem for complicated apps (apps which needs hot update I guess are mainly also large), since they already use a large portion of the framework. Anyway just write down my thoughts about this potential problem.

**Comment by fzyzcjy** [comment]
Duplicate of https://github.com/ethanblake4/dart_eval/issues/24

**Comment by ethanblake4** [comment]
Moved this to a discussion, but it's actually not a duplicate. https://github.com/ethanblake4/dart_eval/issues/24 is about tree-shaking for code _inside_ dart_eval, not about how it prevents the Dart compiler from tree-shaking code _outside_ dart_eval. TBH, I don't know how it would be possible to fix that, but I'll leave this open for any ideas.
